### PR TITLE
GRPCLB fallback mode.

### PR DESCRIPTION
**Pull request by zhangkun83**
While GrpclbLoadBalancer always starts by sending a GRPCLB request to addresses that NameResolver has provided, the addresses may be application servers that don't implement GRPCLB protocol. In that case, we should start sending application traffic to it.

#1137

### CODE REVIEW

File: grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java [hunk 1]
```diff
@@ -340,19 +347,37 @@ private void updateRetainedTransports() {
...
+          if (lbResponseObserver != this) {
+            return;
+          }
+          directTransport = transportFuture = Futures.immediateFuture(lbTransport);
+          pendingPicksFulfillmentBatch = pendingPicks.createFulfillmentBatch();
+        }
+        pendingPicksFulfillmentBatch.link(
+            new Supplier<ListenableFuture<ClientTransport>>() {
+              @Override
+              public ListenableFuture<ClientTransport> get() {
```
Comment (carl-mastrangelo) [inline grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java]: Futures.immediateFuture?

File: grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java [hunk 2]
```diff
@@ -249,6 +255,7 @@ public void transportShutdown(
     handleError(status.augmentDescription("Transport to LB server closed"));
     synchronized (lock) {
       if (transport == lbTransport) {
+        directTransport = null;
```
Comment (ejona86) [inline grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java]: This seems fine, but you may consider moving this line to `connectToLb()` (possibly above `closed` check) to keep reset logic all in the same place. It's fine whichever place you prefer.

**Comment by ejona86** [comment]
@zhangkun83 LGTM

**Comment by carl-mastrangelo** [comment]
LGTM

File: grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java (hunk 2 above)
Comment (zhangkun83) [inline grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java]: Fair enough. Moved to connectToLb().

**Pull request merged**